include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -158,6 +164,73 @@ struct PaddingToAvoidFalseSharing {
   char padding[ORT_FALSE_SHARING_BYTES];
 };
 
+/* Usage:
+1. In executor, call Start() before profiling and Stop() to get profiled numbers;
+2. Inside thread pool, call LogStart() before interested section and LogEnd... after to log elapsed time;
+3. To extend, just add more events in enum Event before "All", and update GetEventName(...) accordingly;
+4. Note LogStart must pair with either LogEnd or LogEndAndStart, otherwise ORT_ENFORCE will fail;
+5. ThreadPoolProfiler is thread-safe.
+*/
+
+class ThreadPoolProfiler {
+ public:
+  enum ThreadPoolEvent {
+    DISTRIBUTION = 0,
+    DISTRIBUTION_ENQUEUE,
+    RUN,
+    WAIT,
+    WAIT_REVOKE,
+    MAX_EVENT
```

Comment:
Unless we are using MAX_EVENT as an ID, I'd normally call this NUM_EVENTS, meaning that the valid IDs will run 0 up to NUM_EVENTS-1 inclusive (for a total of NUM_EVENTS), and use MAX_... to refer to the maximum valid ID.